src/pkg/cluster/state.go

```diff
@@ -154,17 +159,21 @@ func (c *Cluster) InitZarfState(ctx context.Context, initOptions types.ZarfInitO
 		initOptions.ArtifactServer.FillInEmptyValues()
 		state.ArtifactServer = initOptions.ArtifactServer
 	} else {
+		// TODO (@austinabro321) validate immediately in `zarf init` if these are set and not equal and error out if so
```

Comment:
👍 Worth a follow up issue imo. I don't think this would fall under the packager changes so maybe a refactor would be in scope for that issue. I see a lot of deeply nested code which makes it difficult to reason about.

Reply:
Definitely - https://github.com/zarf-dev/zarf/issues/3188